core/ledger/kvledger/txmgmt/statedb/statecouchdb/couchdb_test.go

```diff
@@ -348,97 +306,104 @@ func TestDBBadDatabaseName(t *testing.T) {
 }
 
 func TestDBBadConnection(t *testing.T) {
-	startCouchDB()
 	//create a new instance and database object
 	//Limit the maxRetriesOnStartup to 3 in order to reduce time for the failure
-	config := testConfig()
-	config.Address = badConnectURL
-	config.MaxRetriesOnStartup = 3
-	_, err := CreateCouchInstance(config, &disabled.Provider{})
+	config := &ledger.CouchDBConfig{
+		Address:             badConnectURL,
+		Username:            "",
+		Password:            "",
+		MaxRetries:          3,
+		MaxRetriesOnStartup: 3,
+		RequestTimeout:      35 * time.Second,
+	}
+	_, err := createCouchInstance(config, &disabled.Provider{})
 	assert.Error(t, err, "Error should have been thrown for a bad connection")
 }
 
 func TestBadDBCredentials(t *testing.T) {
-	startCouchDB()
-	database := "testdbbadcredentials"
-	err := cleanup(database)
-	assert.NoError(t, err, "Error when trying to cleanup  Error: %s", err)
-	defer cleanup(database)
+	env := testEnv
+	env.init(t, nil)
+	defer env.cleanup()
 
-	badConfig := testConfig()
+	badConfig := env.config
 	badConfig.Username = "fred"
 	badConfig.Password = "fred"
 	//create a new instance and database object
-	_, err = CreateCouchInstance(badConfig, &disabled.Provider{})
+	_, err := createCouchInstance(badConfig, &disabled.Provider{})
 	assert.Error(t, err, "Error should have been thrown for bad credentials")
+	env.config.Username = ""
+	env.config.Password = ""
 
 }
 
 func TestDBCreateDatabaseAndPersist(t *testing.T) {
-	startCouchDB()
+	env := testEnv
+	env.init(t, nil)
+	config := env.config
 
 	//Test create and persist with default configured maxRetries
-	testDBCreateDatabaseAndPersist(t, testConfig().MaxRetries)
+	testDBCreateDatabaseAndPersist(t, env.config)
+	env.cleanup()
 
 	//Test create and persist with 0 retries
-	testDBCreateDatabaseAndPersist(t, 0)
+	config.MaxRetries = 0
+	testDBCreateDatabaseAndPersist(t, config)
+	env.cleanup()
 
 	//Test batch operations with default configured maxRetries
-	testBatchBatchOperations(t, testConfig().MaxRetries)
+	testBatchBatchOperations(t, env.config)
+	env.cleanup()
 
 	//Test batch operations with 0 retries
-	testBatchBatchOperations(t, 0)
+	testBatchBatchOperations(t, config)
+	env.cleanup()
```

Comment:
Calling env.cleanup() multiple times is a source of confusion. In any case, it is not needed in this test because the deletion of db is explicitly being done in the actual test code (such as "testDBCreateDatabaseAndPersist"). The existing code for this test, the db is not being deleted explicitly. 
So, to keep it uniform with other tests, I would simply call "defer env.cleanup"

Reply:
Earlier, the `cleanup()` was a function and now it is a method on `testEnv`. Hence, the earlier code called `defer cleanup()` on the respective test function. Given that we have moved to `env`, I thought doing the cleanup at a single place is the correct way to do rather than passing around the `env` to each function. If you insist on passing the env, I will do it in a separate PR and let's not get blocked on this.